crud/src/main/java/com/redhat/lightblue/crud/BulkRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.crud;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+
+import com.redhat.lightblue.Request;
+
+/**
+ * Contains a list of requests. Each request has a sequence number
+ * that can be used to match the corresponding response.
+ * <pre>
+ *   {
+ *     "requests": [
+ *         {
+ *             "seq":0,
+ *             "op": "FIND",
+ *             "request": { request }
+ *         }
+ *     ]
+ *   }
+ * </pre> 
+ */
+public class BulkRequest extends AbstractBulkJsonObject<Request> {
+
+    @Override
+    public JsonNode toJson() {
+        return toJson("requests","request");
```

Comment:
That's what the base class method does.

On Thu, Aug 27, 2015 at 11:54 AM, Dennis Crissman <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/crud/BulkRequest.java
> <https://github.com/lightblue-platform/lightblue-core/pull/437#discussion_r38125191>
> :
>
> > + *   {
> > + *     "requests": [
> > + *         {
> > + *             "seq":0,
> > + *             "op": "FIND",
> > + *             "request": { request }
> > + *         }
> > + *     ]
> > + *   }
> > + * </pre>
> > + */
> > +public class BulkRequest extends AbstractBulkJsonObject<Request> {
> > +
> > +    @Override
> > +    public JsonNode toJson() {
> > +        return toJson("requests","request");
>
> Why not iterate over entries and call their toJson methods?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/437/files#r38125191>
> .
>